src/tools/compiletest/src/directives/cfg.rs

```diff
@@ -74,230 +80,181 @@ fn parse_cfg_name_directive<'a>(
     // FIXME(Zalathar): This currently allows either a space or a colon, and
     // treats any "value" after a colon as though it were a remark.
     // We should instead forbid the colon syntax for these directives.
-    let comment = line.remark_after_space().or_else(|| line.value_after_colon());
-
-    // Some of the matchers might be "" depending on what the target information is. To avoid
-    // problems we outright reject empty directives.
-    if name.is_empty() {
-        return ParsedNameDirective::not_handled_here();
+    let comment = line
+        .remark_after_space()
+        .or_else(|| line.value_after_colon())
+        .map(|c| c.trim().trim_start_matches('-').trim());
+
+    if let Some(cond) = conditions.conds.get(name) {
+        ParsedNameDirective {
+            pretty_reason: Some(Arc::clone(&cond.message_when_ignored)),
+            comment,
+            outcome: if cond.value { MatchOutcome::Match } else { MatchOutcome::NoMatch },
+        }
+    } else {
+        ParsedNameDirective { pretty_reason: None, comment, outcome: MatchOutcome::Invalid }
     }
+}
 
-    let mut outcome = MatchOutcome::Invalid;
-    let mut message = None;
-
-    macro_rules! condition {
-        (
-            name: $name:expr,
-            $(allowed_names: $allowed_names:expr,)?
-            $(condition: $condition:expr,)?
-            message: $($message:tt)*
-        ) => {{
-            // This is not inlined to avoid problems with macro repetitions.
-            let format_message = || format!($($message)*);
-
-            if outcome != MatchOutcome::Invalid {
-                // Ignore all other matches if we already found one
-            } else if $name.custom_matches(name) {
-                message = Some(format_message());
-                if true $(&& $condition)? {
-                    outcome = MatchOutcome::Match;
-                } else {
-                    outcome = MatchOutcome::NoMatch;
-                }
-            }
-            $(else if $allowed_names.custom_contains(name) {
-                message = Some(format_message());
-                outcome = MatchOutcome::NoMatch;
-            })?
-        }};
-    }
+/// Uses information about the current target (and all targets) to pre-compute
+/// a value (true or false) for a number of "conditions". Those conditions can
+/// then be used by `ignore-*` and `only-*` directives.
+pub(crate) fn prepare_conditions(config: &Config) -> PreparedConditions {
```

Comment:
Suggestion: can you add a (duplicate) remark here for locality that the order of preparation/registration matters? In that, earlier prepared/registered condition names will take precedence over later prepared condition names if they ever collide.